src/components/CreateFileDialog.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Fade,
+  LinearProgress,
+  TextField,
+} from "@material-ui/core";
+import { CloudUploadOutlined } from "@material-ui/icons";
+import { CreateFileRequestDataAttributes } from "@vertexvis/api-client-node";
+import React from "react";
+
+import { CreateFileRes } from "../pages/api/files";
+
+interface CreateFileDialogProps {
+  readonly open: boolean;
+  readonly onClose: () => void;
+  readonly onFileCreated: (fileId: string) => void;
+}
+
+export default function CreateFileDialog({
+  open,
+  onClose,
+  onFileCreated,
+}: CreateFileDialogProps): JSX.Element {
+  const [file, setFile] = React.useState<File | null | undefined>(null);
+  const [suppliedId, setSuppliedId] = React.useState<string | null>(null);
+  const [rootFileName, setRootFileName] = React.useState<string | null>(null);
```

Comment:
I've been trying to use `undefined` instead. Or do you need it to be null for serialization?

Reply:
ok, I can update it - I'm not including it if it's falsey